### PR TITLE
prov/tcp; Connection management change to remove the thread

### PR DESCRIPTION
**Pull request by vkrishna**
This PR removes the connection manager thread and replace it with event queue based mechanism. Each of the commit has description about what it is trying to achieve.

> Event queue's wait object is used for monitoring the endpoints for connection events. 
> EQ read progresses the connection setup
> changed some names relevantly.

### CODE REVIEW

File: prov/tcp/src/tcpx_conn_mgr.c [hunk 1]
```diff
@@ -474,171 +285,147 @@ static void handle_conn_handle(struct poll_fd_mgr *poll_mgr,
...
-	fd_signal_free(&tcpx_fabric->poll_mgr.signal);
-err:
-	fastlock_destroy(&tcpx_fabric->poll_mgr.lock);
-	return ret;
+	num_fds = fi_epoll_wait(wait_fd->epoll_fd, wait_contexts,
+				MAX_EPOLL_EVENTS, 0);
+	if (num_fds < 0)
+		return;
+
+	for ( i = 0; i < num_fds; i++)
```
Comment (sydidelot) [inline prov/tcp/src/tcpx_conn_mgr.c]: @vkrishna While testing your new code, I discovered an issue in the function tcpx_conn_mgr_run().
After fi_epoll_wait() returns, you need to make sure that wakeup signals do not enter process_cm_ctx() and must be skipped. You can fix this with the following lines:

```
       for ( i = 0; i < num_fds; i++) {
               /* Skip wake up signals */
               if (&wait_fd->util_wait.wait_fid.fid == wait_contexts[i])
                       continue;

                process_cm_ctx(eq->wait,
                                 (struct tcpx_cm_context *)
                                 wait_contexts[i]);
       }
```

Comment (vkrishna) [reply prov/tcp/src/tcpx_conn_mgr.c]: Thanks @sydidelot for taking time checking the patch and fixing this issue. I will add it.

**Comment by dmitrygladkov** [comment]
Would you like me to test this PR?

**Comment by shefty** [comment]
Can you please refresh the PR to resolve conflicts?

**Comment by vkrishna** [comment]
Resolved the conflicts with base branch.

File: prov/tcp/src/tcpx_conn_mgr.c [hunk 2]
```diff
@@ -114,6 +114,325 @@ static int tcpx_ep_msg_xfer_enable(struct tcpx_ep *ep)
...
+	return ret;
+}
+
+int tcpx_eq_wait_try_func(void *arg)
+{
+	return FI_SUCCESS;
+}
+
+static void client_wait_for_connresp(struct util_wait *wait,
+				     struct poll_fd_info *poll_info)
```
Comment (shefty) [inline prov/tcp/src/tcpx_conn_mgr.c]: AFAICT, this call doesn't actually wait for a response.  If that's the case, please rename the call.  If the call is actually blocking until the server sends a response, then we have other issues and need to change that behavior. :)

**Comment by shefty** [comment]
I left a comment on one of the patches.  Overall, this seems reasonable and looks like a good change.  Merging -- additional changes can come as follow up patches.  Thanks!

**Pull request merged**